src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -539,26 +553,23 @@ public void putKeywords(List<String> keywords) {
      * @param keyword Keyword to add
      */
     public void addKeyword(String keyword) {
-        Objects.requireNonNull(keyword, "keyword must not be empty");
+        Objects.requireNonNull(keyword, "keyword must not be null");
 
         if (keyword.isEmpty()) {
             return;
         }
 
-        List<String> keywords = this.getSeparatedKeywords();
-        Boolean duplicate = false;
+        List<String> keywords = this.getKeywords();
 
         for (String key : keywords) {
             if (keyword.equalsIgnoreCase(key)) {
-                duplicate = true;
-                break;
+                // Duplicate so don't add it
+                return;
             }
         }
 
-        if (!duplicate) {
-            keywords.add(keyword);
-            this.putKeywords(keywords);
-        }
+        keywords.add(keyword);
```

Comment:
If you don't want keyword multiple times, then simply use a Set, which by default does not allow double values.
Then you could remove the check.